src/gui/attributetable/qgsvectorlayerselectionmanager.h

```diff
@@ -42,6 +42,14 @@ class GUI_EXPORT QgsVectorLayerSelectionManager : public QgsIFeatureSelectionMan
     void setSelectedFeatures( const QgsFeatureIds &ids ) override;
     const QgsFeatureIds &selectedFeatureIds() const override;
 
+    /**
+     * Returns the vector layer
+     */
+    QgsVectorLayer *layer() const;
+
+  private slots:
+    virtual void onSelectionChanged( const QgsFeatureIds &selected, const QgsFeatureIds &deselected, bool clearAndSelect );
```

Comment:
Can you add a docstring to that explaining default behavior and specs for overriding?

Reply:
Done